### ISSUE TITLE
yabai process restarting even after stopping the service

### ISSUE THREAD

**Issue by anujc4** [koekeishiya/yabai#1921]
tldr; Not able to stop yabai process once started

## Expected Behaviour

When I perform `yabai --stop-service`, I should not see a yabai process anymore

## Current Behaviour

Yabai restarts itself with a different PID

## Steps to reproduce

I don't know if this is a wide-known issue, but here is how I reproduced the issue. 

1. Run `yabai --stop-service`
2. Run `ps aux |  grep yabai`, the yabai process still exists
3. Try to manually kill the process `killall yabai`
4. Run `ps aux |  grep yabai`, the yabai process will still be there with a different PID

This is the output from my terminal

<img width="1895" alt="image" src="https://github.com/koekeishiya/yabai/assets/9458914/3e43ce05-5cbb-4632-8d37-e88580837b67">

## Context

I am using the release version 6.0.0

```
$ yabai --version
yabai-v6.0.0
```

**Comment by Odaym** [comment]
what the hell did you do to my yabai! ever since the last update everything is messed up

**Comment by ucasyfp** [comment]
Same here, but I am using version 5.0.9.

**Comment by anujc4** [comment]
> what the hell did you do to my yabai! ever since the last update everything is messed up

Calm down mate, Open Source is hard and it's difficult for one developer (huge props to @koekeishiya btw) to maintain a tool that is so widely used and loved. I'm pretty sure this is just a minor bug and won't even effect any workflows in a breaking way.

**Comment by Odaym** [comment]
I'm sorry, I just love yabai too much!

**Comment by prurph** [comment]
I'm unable to repro this on yabai-v6.0.1 macOS Sonoma 14.1.1.
